### PR TITLE
Fix deregister script a lot

### PR DESCRIPTION
**Pull request by tthyer**
deregister script was seriously borked, fixed it all up

### CODE REVIEW

**Review by thomasyu888 — APPROVED** [review]

File: deregister_ami.sh [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
   # deregister the image
-  aws ec2 deregister-image --image-id $AMI_ID;
-  sleep 2
+  aws --profile "$AWS_PROFILE" --region "$AWS_REGION" ec2 deregister-image --image-id "$AMI_ID"
+else
+  echo "No AMI ID for $AMI_NAME"
+  exit 1
 fi
-
+sleep 10
```
Comment (zaro0508) [inline deregister_ami.sh]: why do we need this sleep?

File: deregister_ami.sh [hunk 2]
```diff
@@ -1,26 +1,42 @@
 ###############################################
-# Remove an AMI and it's associated snapshots #
+# Remove an AMI and its associated snapshots #
 ###############################################
+set -xe
```
Comment (zaro0508) [inline deregister_ami.sh]: too noisy with this, can we remove?

File: deregister_ami.sh [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
-for snap in "${SNAPS[@]}";
+for snap in "${SNAPS[@]}"
 do
-  echo "delete snapshot: $snap";
-  aws ec2 delete-snapshot --snapshot-id $snap;
+  echo "delete snapshot: $snap"
+  aws --profile "$AWS_PROFILE" --region "$AWS_REGION" ec2 delete-snapshot --snapshot-id $snap
 done
+
+set +xe
```
Comment (zaro0508) [inline deregister_ami.sh]: remove?

**Review by zaro0508 — APPROVED** [review]
thanks for fixing.  i could have sworn this script worked for me in the past.

**Pull request merged**